tikv/mvcc.go

```diff
@@ -639,15 +639,30 @@ func (store *MVCCStore) Commit(req *requestCtx, keys [][]byte, startTS, commitTS
 	var tmpDiff int
 	var isPessimisticTxn bool
 	for _, key := range keys {
+		var lockErr error
+		var checkErr error
+		var lock mvcc.MvccLock
 		buf = store.lockStore.Get(key, buf)
 		if len(buf) == 0 {
 			// We never commit partial keys in Commit request, so if one lock is not found,
 			// the others keys must not be found too.
-			return store.handleLockNotFound(req, key, startTS, commitTS)
+			lockErr = ErrLockNotFound
+		} else {
+			lock = mvcc.DecodeLock(buf)
+			if lock.StartTS != startTS {
+				lockErr = ErrReplaced
+			}
 		}
-		lock := mvcc.DecodeLock(buf)
-		if lock.StartTS != startTS {
-			return ErrReplaced
+		if lockErr != nil {
+			// Maybe the secondary keys committed by other concurrent transactions using lock resolver,
+			// check commit info from store
+			checkErr = store.handleLockNotFound(req, key, startTS, commitTS)
+			if checkErr == nil {
+				continue
```

Comment:
We can return here because the transaction is already committed.

Reply:
The commit batch keys may be partially resolved by other transactions, commit batch contains secondary keys "a, b, c",  "b" is  resolved and committed by other transactions, "a, c" may still be waiting for commiting operations